operator/api/v1alpha1/kyma_types.go

```diff
@@ -20,24 +20,92 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
-// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+// ComponentType defines the components to be installed
+type ComponentType struct {
+	Name     string            `json:"name"`
+	Settings map[string]string `json:"settings"`
+}
 
 // KymaSpec defines the desired state of Kyma
 type KymaSpec struct {
-	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
-	// Important: Run "make" to regenerate code after modifying this file
-
-	// Foo is an example field of Kyma. Edit kyma_types.go to remove/update
-	Foo string `json:"foo,omitempty"`
+	// Components specifies the list of components to be installed
+	Components []ComponentType `json:"components,omitempty"`
 }
 
 // KymaStatus defines the observed state of Kyma
 type KymaStatus struct {
-	// INSERT ADDITIONAL STATUS FIELD - define observed state of cluster
-	// Important: Run "make" to regenerate code after modifying this file
+	// State signifies current state of Kyma.
+	// Value can be one of ("Ready", "Processing", "Error", "Deleting").
+	State KymaState `json:"state,omitempty"`
+
+	// List of status conditions to indicate the status of a ServiceInstance.
+	// +optional
+	Conditions []KymaCondition `json:"conditions,omitempty"`
+
+	// Observed generation
+	// +optional
+	ObservedGeneration int64 `json:"observedGeneration,omitempty"`
+}
+
+type KymaState string
+
+// Valid Kyma States
+const (
+	// KymaStateReady signifies Kyma is ready
+	KymaStateReady KymaState = "Ready"
+
+	// KymaStateProcessing signifies Kyma is reconciling
+	KymaStateProcessing KymaState = "Processing"
```

Comment:
Why do we need custom states here instead of reusing the states from Apimachinery (e.g. like Deployment with "Progressing")